tool/src/js/mod.rs

```diff
@@ -75,24 +113,43 @@ pub fn gen_bindings(env: &Env, outs: &mut HashMap<String, String>) -> fmt::Resul
 /// a struct, at the boundary as well.
 fn used_types<'env>(
     custom_type: &'env ast::CustomType,
+    is_declaration: bool,
     in_path: &ast::Path,
     env: &'env Env,
 ) -> HashSet<&'env ast::CustomType> {
     let mut set = HashSet::new();
+    let cfg = if is_declaration {
+        UsedTypesCfg::DeclarationFile
+    } else {
+        UsedTypesCfg::ImplAll
+    };
 
     if let ast::CustomType::Struct(strct) = custom_type {
         for (_, typ, _) in strct.fields.iter() {
-            used_types_inner(typ, &mut set, in_path, env, false);
+            used_types_inner(typ, &mut set, in_path, env, cfg);
         }
     }
 
     for method in custom_type.methods() {
-        for param in method.params.iter() {
-            used_types_inner(&param.ty, &mut set, in_path, env, false);
+        if is_declaration {
+            for param in method.params.iter() {
+                used_types_inner(&param.ty, &mut set, in_path, env, cfg);
+            }
+        } else {
+            // We only want the enums because we have to convert those
+            for param in method.params.iter() {
+                used_types_inner(
+                    &param.ty,
+                    &mut set,
+                    in_path,
+                    env,
+                    UsedTypesCfg::ImplEnumOnly,
+                );
+            }
```

Comment:
Suggestion (optional):

```suggestion
        // We only want the enums because we have to convert those
        let cfg_inner = if is_declaration { cfg } else { UsedTypesCfg::ImplEnumOnly };
        for param in method.params.iter() {
            used_types_inner(&param.ty, &mut set, in_path, env, cfg_inner);
        }
```